src/renderer/components/message/MessageList.tsx

```diff
@@ -92,13 +92,21 @@ function useUnreadCount(
   return freshMessageCounter
 }
 
+type MessageListProps = {
+  chatStore: ChatStoreStateWithChatSet
+  refComposer: todo
+  selectMessage: (id: number) => void
+  unselectMessage: (id: number) => void
+  selectedMessages: number[]
```

Comment:
`MessageList` and `Message` get quite many new props here which I wouldn't see _directly_ as part of the message component itself.

Maybe we can introduce a context here instead to manage the selection state for messages? The context lives in the currently displayed chat and every message can consume the state of the context to understand if it is selected or not.

I've seen you already have a nice reducer. With the context representing the state and the reducer changing it we can have a nice isolated logic. All of this can be moved into own files.